src/Device.js

```diff
@@ -38,6 +39,52 @@ class Device extends Component {
     this.toggleInfo = this.toggleInfo.bind(this)
   }
 
+  renderAppVersionSuggestion = (installed, suggested) => {
+    return (
+      <table style={{ width: 'auto' }}>
+        <tbody>
+          <tr>
+            <td>Suggested version:</td>
+            <td>
+              <span className='suggested-value'>{String(suggested)}</span>
+            </td>
+          </tr>
+          <tr>
+            <td>Your version:</td>
+            <td>
+              <span className='suggested-value'>{String(installed)}</span>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    )
+  }
+
+  renderApplicationStateMessage = ({ state, version, installed, policy = {} }) => {
+    if (state === 'INVALID_INSTALL_STATE' && !installed) {
+      const installPrompt = policy.installFrom ? (
+        <p>
+          {' '}
+          Install from{' '}
+          <a href={policy.installFrom} target='_blank'>
```

Comment:
small nit, should have `rel="noopener noreferrer"` to prevent compilation warning